### DISCUSSION TITLE
How to set .env location in deno's edge functions test

### DISCUSSION THREAD

**Discussion by TalMikey** [supabase/supabase#27140] [Questions]
I try to test my edge functions but fail to access .env file.

The steps:

1. I created a test file in the location: `supabase\functions\tests\try.test.ts`
2. Created both `.env` and `.env.local` files in the root directory (same level of supabase directory) and also in the tests directory. 
3. The test:
```
Deno.test("try", async () => {
   console.log({url: Deno.env.get('SUPABASE_URL') });
})
```
4. I run `deno test --allow-all ../supabase/functions/tests` and `deno test --allow-all ../supabase/functions/tests -- --env-file .env.local`
5. The output:
```
------- output -------
{ url: undefined }
----- output end -----
try ... ok (12ms)

ok | 1 passed | 0 failed (44ms)
```

As seen, I can't access the .env file.

**Comment by khoaln** [comment]
I'm using dotenv https://deno.land/std@0.224.0/dotenv/mod.ts to load variables from .env.local. You can also create a test setup and call Deno.env.set to set the variables loaded from .env.local

**Comment by gc-ft** [comment]
.env should be in functions directory for supabase to pick it up.
